src/bidiMapper/domains/network/networkRequest.ts

```diff
@@ -0,0 +1,365 @@
+
+  #sendBeforeRequestEvent() {
+    this.#eventManager.registerEvent(
+      this.#getBeforeRequestEvent(),
+      this.#requestWillBeSentEvent?.frameId ?? null
+    );
+  }
+
+  #getBeforeRequestEvent(): Network.BeforeRequestSentEvent {
+    if (this.#requestWillBeSentEvent === undefined) {
+      throw new Error('RequestWillBeSentEvent is not set');
+    }
+    if (this.#requestWillBeSentExtraInfoEvent === undefined) {
+      throw new Error('RequestWillBeSentExtraInfoEvent is not set');
+    }
+
+    const requestWillBeSentEvent = this.#requestWillBeSentEvent!;
+    const requestWillBeSentExtraInfoEvent =
+      this.#requestWillBeSentExtraInfoEvent!;
+
+    const baseEventParams = {
+      context: requestWillBeSentEvent.frameId ?? null,
+      navigation: requestWillBeSentEvent.loaderId,
+      // TODO: implement.
+      redirectCount: 0,
+      request: {
+        request: requestWillBeSentEvent.requestId,
+        url: requestWillBeSentEvent.request.url,
+        method: requestWillBeSentEvent.request.method,
+        headers: Object.keys(requestWillBeSentEvent.request.headers).map(
+          (key) => ({
+            name: key,
+            value: requestWillBeSentEvent.request.headers[key],
+          })
+        ),
+        cookies: NetworkRequest.#getCookies(
+          requestWillBeSentExtraInfoEvent.associatedCookies
+        ),
+        // TODO: implement.
+        headersSize: -1,
+        // TODO: implement.
+        bodySize: 0,
+        timings: {
+          // TODO: implement.
+          timeOrigin: 0,
+          // TODO: implement.
+          requestTime: 0,
+          // TODO: implement.
+          redirectStart: 0,
+          // TODO: implement.
+          redirectEnd: 0,
+          // TODO: implement.
+          fetchStart: 0,
+          // TODO: implement.
+          dnsStart: 0,
+          // TODO: implement.
+          dnsEnd: 0,
+          // TODO: implement.
+          connectStart: 0,
+          // TODO: implement.
+          connectEnd: 0,
+          // TODO: implement.
+          tlsStart: 0,
+          // TODO: implement.
+          tlsEnd: 0,
+          // TODO: implement.
+          requestStart: 0,
+          // TODO: implement.
+          responseStart: 0,
+          // TODO: implement.
+          responseEnd: 0,
+        },
+      },
+      // Timestamp should be in milliseconds, while CDP provides it in seconds.
+      timestamp: Math.round(requestWillBeSentEvent.wallTime * 1000),
+    };
+
+    const params: Network.BeforeRequestSentParams = {
+      ...baseEventParams,
+      initiator: {type: this.#getInitiatorType()},
+    };
+    return {
+      method: Network.EventNames.BeforeRequestSentEvent,
+      params,
+    };
+  }
+
+  #getInitiatorType(): 'parser' | 'script' | 'preflight' | 'other' {
+    switch (this.#requestWillBeSentEvent?.initiator.type) {
+      case 'parser':
+        return 'parser';
+      case 'script':
+        return 'script';
+      case 'preflight':
+        return 'preflight';
```

Comment:
Optionally:

```
case 'parser':
case 'script':
case 'preflight':
  return this.#requestWillBeSentEvent?.initiator.type;
default:
  return 'other';
```

Only if you think it's more readable and/or less error-prone.